contrib/dhcp_agent/dhcp_agent.py

```diff
@@ -0,0 +1,351 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import atexit
+import errno
+import fcntl
+import hashlib
+import logging
+import os
+import subprocess
+import sys
+from optparse import OptionParser
+from logging import handlers as logging_handlers
+
+IS_PY3 = sys.version_info[0] == 3
+if IS_PY3:
+    from dbm import gnu as gdbm
+    from urllib.request import urlopen, Request
+    from urllib.parse import urlencode
+    from urllib.error import HTTPError
+else:
+    import gdbm
+    from urllib import urlencode
+    from urllib2 import urlopen, Request, HTTPError
+
+
+MODES = ('all', 'networks', 'entries')
+MODE_ALL, MODE_NETWORKS, MODE_ENTRIES = MODES
+
+PROTOS = ('http', 'https')
+PROTO_HTTP, PROTO_HTTPS = PROTOS
+
+CONFIG_HTTP_LAST_MODIFIED = 'http-last-modified'
+
+
+def _get_cmd_parser():
+    parser = OptionParser(
+        description='Update configuration in DHCP server.',
+    )
+    parser.add_option('-H', '--host', help='Ralph instance host.')
+    parser.add_option('-k', '--key', help='Ralph API key.')
+    parser.add_option(
+        '-m',
+        '--mode',
+        type='choice',
+        choices=MODES,
+        default=MODE_ALL,
+        help='Choose what part of config you want to upgrade. '
+             '[Default: all; Options: {}]'.format(', '.join(MODES)),
+    )
+    parser.add_option(
+        '-l',
+        '--log-path',
+        help='Path to log file. [Default: STDOUT]',
+        default='STDOUT',
+    )
+    parser.add_option(
+        '-c',
+        '--dhcp-config-entries',
+        help='Path to the DHCP entries configuration file.',
+    )
+    parser.add_option(
+        '-n',
+        '--dhcp-config-networks',
+        help='Path to the DHCP networks configuration file.',
+    )
+    parser.add_option(
+        '-a',
+        '--proto',
+        type='choice',
+        choices=PROTOS,
+        default=PROTO_HTTPS
+    )
+    parser.add_option(
+        '-e',
+        '--net-env',
+        help='Only get config for the specified environment.',
```

Comment:
It's strongly connected with ``NetworkEnvironment`` in Ralph3. Environment is some group of networks (see https://github.com/ar4s/ralph/blob/ng-dhcp-agent/src/ralph/networks/models/networks.py#L36 and https://github.com/ar4s/ralph/blob/ng-dhcp-agent/src/ralph/networks/models/networks.py#L148).